test/boost/commitlog_test.cc

```diff
@@ -803,13 +801,11 @@ SEASTAR_TEST_CASE(test_commitlog_deadlock_in_recycle) {
         }
     } catch (timed_out_error&) {
         BOOST_FAIL("log write timed out. maybe it is deadlocked... Will not free log. ASAN errors and leaks will follow...");
-        release = false;
+        abort();
```

Comment:
Doesn't `BOOST_FAIL()` throw? I think this code is unreachable, as was the code it replaces.

Reply:
Does it throw? I thought it just set error condition + logged? Seemingly, we'd run the code after the fails in the tests runs I cannot repro, but maybe they were actually executing raii clauses...

Reply:
You are right, `REQUIRE` will abort-throw. Would perhaps be nice if test harness could treat this (optionally) as ::abort() point instead...
@bhalevy - since it is you requesting this, should I change this to message + ::abort() or should we just clean away pretend cleanup-clauses and look at having abort as a function of failed REQUIRE?

Reply:
I think reporting to testlog.error then ::abort would be the simplest way

Reply:
Changed.